typescript/tuple_tree.ts

```diff
@@ -311,8 +315,8 @@ function setComm<T>(set1: Set<T>, set2: Set<T>): Comm<T> {
     return { only1: only1, only2: only2, common: common };
 }
 
-function hasCustomJson(obj: any): obj is { toJSON: () => any } {
-    return typeof obj === "object" && "toJSON" in obj && typeof obj.toJSON === "function";
+function hasComparisonObject(obj: any): obj is { comparisonObject: () => any } {
+    return typeof obj === "object" && typeof obj?.comparisonObject === "function";
 }
```

Comment:
1. What is a "comparison object"? A comparator?
2. Do  we really have to use this kind of introspection? Introspection for this kind of stuff usually means there could be better software engineering somewhere else.

Reply:
Before `comparisonObject`, `toJSON` was used to compare a couple of things (namely String-derivates and `MetaAddress`) since the string representation of those was stable and could be compared for equality. Since now most model objects have toJSON, another name was used.

Reply:
Added `.equals`